### DISCUSSION TITLE
Making cURL POST requests to interact with GraphQL in cmd on Windows 11

### DISCUSSION THREAD

**Discussion by arielwu415** [curl/curl#11309] [Q&A] [answered]
Hi cURL community,

I'm currently facing issues of making Shopify GraphQL Admin API POST requests through curl in cmd on Windows 11.
I need help with figuring out why this command is not working on Windows, and what I could do to make it work.

I've counseled [Shopify dev support](https://community.shopify.com/c/shopify-apis-and-sdks/using-curl-command-to-add-metaobject-entries-graphql-admin-api/m-p/2077884), and they refer me to ask cURL community as the command works in their environment (Mac OS) but not mine. It would be great if anyone could let me know what I do wrong here.

Here's the command I entered in cmd (all in the same line)
```
curl -X POST https://shop-name.myshopify.com/admin/api/2023-04/graphql.json
-H 'Content-Type:application/graphql'
-H 'X-Shopify-Access-Token:my-access-token'
-d '{"query":"{products(first:3){edges{node{id}}}}"}'
```
Here's the error I got
```
{"errors":{"query":"Required parameter missing or invalid"}}
```

Any input is appreciated. Thank you for your time reading this.

**Comment by bagder** [comment, marked as answer]
Because Windows shells don't support single-quotes like that on the command line?

**Reply by jay** [reply]
Yes, use double quotes and escape inner double quotes like this
`"{\"query\":\"{products(first:3){edges{node{id}}}}\"}"`

**Comment by bagder** [comment]
Also, but unrelated: [unnecessary use of -X](https://daniel.haxx.se/blog/2015/09/11/unnecessary-use-of-curl-x/)
